### PR TITLE
fix: switch from pooling to default method to detect changes

### PR DESCRIPTION
**Pull request by mykalmax**
fixing https://github.com/TobikoData/sqlmesh/issues/3109

so when using `pooling` (and pooling is not default option for wathfiles). It seems to treat some files (like symlinks for example) as deleted and just keeps emitting deletion change. And that results in UI keep triggering updates and API calls (an infinite loop)

### CODE REVIEW

File: web/server/watcher.py [hunk 1]
```diff
@@ -28,33 +28,41 @@ async def watch_project() -> None:
         (settings.project_path / c.METRICS).resolve(),
         (settings.project_path / c.SEEDS).resolve(),
     ]
-    ignore_entity_patterns = context.config.ignore_patterns if context else c.IGNORE_PATTERNS
-    ignore_entity_patterns.append("^\\.DS_Store$")
-    ignore_entity_patterns.append("^.*\\.db(\\.wal)?$")
-    ignore_paths = [str((settings.project_path / c.CACHE).resolve())]
-    watch_filter = DefaultFilter(
-        ignore_paths=ignore_paths, ignore_entity_patterns=ignore_entity_patterns
+    watch_filter = DefaultFilter()
```
Comment (mykalmax) [inline web/server/watcher.py:31]: here i wanted to extend defaults

File: web/client/src/library/components/fileExplorer/context.tsx [hunk 2]
```diff
@@ -158,7 +158,7 @@ export default function FileExplorerProvider({
   }
```
Comment (mykalmax) [inline web/client/src/library/components/fileExplorer/context.tsx:159]: path should not contain leading  `/` and name should not contain `/` at all

File: web/client/src/library/pages/root/Navigation.tsx [hunk 3]
```diff
@@ -14,7 +14,7 @@ export default function PageNavigation(): JSX.Element {
   return (
     <div
       className={clsx(
-        'relative min-w-[10rem] px-2 min-h-8 max-h-8 w-full flex items-center',
+        'relative min-w-[10rem] px-2 min-h-[2rem] max-h-[2rem] w-full flex items-center',
```
Comment (mykalmax) [inline web/client/src/library/pages/root/Navigation.tsx:17]: fixing small layout shift when showing `loading` message

File: web/server/watcher.py (hunk 1 above)
Comment (vchan) [reply web/server/watcher.py:31]: I personally would have done it the following way:
```python
DefaultFilter(
    ignore_paths=DefaultFilter.ignore_paths + ignore_paths,
    ignore_entity_patterns=DefaultFilter.ignore_entity_patterns + ignore_entity_patterns,
)
```

**Review by vchan — APPROVED** [review]

File: web/server/watcher.py [hunk 4]
```diff
@@ -28,33 +28,41 @@ async def watch_project() -> None:
...
-                elif change == Change.deleted or not path.exists():
+                if change == Change.deleted or not path.exists():
                     changes.append(
                         models.ArtifactChange(
                             change=Change.deleted,
                             path=str(relative_path),
                         )
                     )
-                elif change == Change.modified and path.is_file():
+                if change == Change.added:
```
Comment (vchan) [inline web/server/watcher.py:59]: Not `elif`?

File: web/server/watcher.py [hunk 5]
```diff
@@ -28,33 +28,40 @@ async def watch_project() -> None:
         (settings.project_path / c.METRICS).resolve(),
         (settings.project_path / c.SEEDS).resolve(),
     ]
+    ignore_dirs = ensure_list(".env")
```
Comment (vchan) [inline web/server/watcher.py:31]: This doesn't seem necessary when you can explicitly make this a list, i.e. `ignore_dirs = [".env"]`.

File: web/server/utils.py [hunk 6]
```diff
@@ -104,3 +104,22 @@ def is_relative_to(path: PurePath, other: PurePath | str) -> bool:
         return True
     except ValueError:
         return False
+
+
+def ensure_list(value: t.Optional[t.Any] = None) -> t.List:
```
Comment (vchan) [inline web/server/utils.py:109]: Can we use ensure_list from from sqlglot.helper? I see it used in various places in our codebase.

**Review by vchan — APPROVED** [review]

**Pull request merged**